studio/components/interfaces/TableGridEditor/DeleteConfirmationDialogs.tsx

```diff
@@ -0,0 +1,241 @@
+import { useQueryClient } from '@tanstack/react-query'
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import ConfirmationModal from 'components/ui/ConfirmationModal'
+import { entityTypeKeys } from 'data/entity-types/keys'
+import { sqlKeys } from 'data/sql/keys'
+import { tableKeys } from 'data/tables/keys'
+import { useGetTables } from 'data/tables/tables-query'
+import { useStore, useUrlState } from 'hooks'
+import { TableLike } from 'hooks/misc/useTable'
+import { noop } from 'lib/void'
+import { useTableEditorStateSnapshot } from 'state/table-editor'
+import { Alert, Button, Checkbox, IconExternalLink, Modal } from 'ui'
+
+export type DeleteConfirmationDialogsProps = {
+  projectRef?: string
+  selectedTable?: TableLike
+  onAfterDeleteTable?: (tables: TableLike[]) => void
+}
+
+const DeleteConfirmationDialogs = ({
+  projectRef,
+  selectedTable,
+  onAfterDeleteTable = noop,
+}: DeleteConfirmationDialogsProps) => {
+  const { project } = useProjectContext()
+  const snap = useTableEditorStateSnapshot()
+
+  const router = useRouter()
+  const [, setParams] = useUrlState({ arrayKeys: ['filter', 'sort'] })
+
+  const queryClient = useQueryClient()
+
+  const { meta, ui } = useStore()
+
+  const getTables = useGetTables({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+
+  const removeDeletedColumnFromFiltersAndSorts = (columnName: string) => {
+    setParams((prevParams) => {
+      const existingFilters = (prevParams?.filter ?? []) as string[]
+      const existingSorts = (prevParams?.sort ?? []) as string[]
+
+      return {
+        ...prevParams,
+        filter: existingFilters.filter((filter: string) => {
+          const [column] = filter.split(':')
+          if (column !== columnName) return filter
+        }),
+        sort: existingSorts.filter((sort: string) => {
+          const [column] = sort.split(':')
+          if (column !== columnName) return sort
+        }),
+      }
+    })
+  }
+
+  const isDeleteWithCascade = snap.confirmationDialog?.isDeleteWithCascade ?? false
+
+  const onConfirmDeleteColumn = async () => {
+    if (!(snap.confirmationDialog?.type === 'column')) return
+
+    const selectedColumnToDelete = snap.confirmationDialog.column
+    try {
+      if (selectedColumnToDelete === undefined) return
+
+      const response: any = await meta.columns.del(selectedColumnToDelete.id, isDeleteWithCascade)
+      if (response.error) throw response.error
+
+      removeDeletedColumnFromFiltersAndSorts(selectedColumnToDelete.name)
+
+      ui.setNotification({
+        category: 'success',
+        message: `Successfully deleted column "${selectedColumnToDelete.name}"`,
+      })
+
+      queryClient.invalidateQueries(sqlKeys.query(project?.ref, ['foreign-key-constraints']))
```

Comment:
I didn't put it inside the `Promise.all` because it wasn't used in this view, so the invalidate call was just marking to be re-fetched next time the query was mounted. However, react-query handles this for us, so I have moved it inside the `Promise.all` 🙌🏻